flink-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBaseTest.java

```diff
@@ -84,111 +90,108 @@ public void testThrowErrorOnClose() throws Exception {
-		TestCassandraSink casSinkFunc = new TestCassandraSink();
+		try (TestCassandraSink casSinkFunc = new TestCassandraSink()) {
+			casSinkFunc.open(new Configuration());
 
-		casSinkFunc.open(new Configuration());
-
-		Exception cause = new RuntimeException();
-		casSinkFunc.setResultFuture(ResultSetFutures.fromCompletableFuture(FutureUtils.getFailedFuture(cause)));
+			Exception cause = new RuntimeException();
+			casSinkFunc.enqueueResultSetFuture(ResultSetFutures.fromCompletableFuture(FutureUtils.getFailedFuture(cause)));
 
-		casSinkFunc.invoke("hello");
+			casSinkFunc.invoke("hello");
 
-		try {
-			casSinkFunc.invoke("world");
-			Assert.fail("Sending of second value should have failed.");
-		} catch (IOException e) {
-			Assert.assertEquals(cause, e.getCause());
-			Assert.assertEquals(0, casSinkFunc.getNumOfPendingRecords());
+			try {
+				casSinkFunc.invoke("world");
+				Assert.fail("Sending of second value should have failed.");
+			} catch (IOException e) {
+				Assert.assertEquals(cause, e.getCause());
+				Assert.assertEquals(0, casSinkFunc.getAcquiredPermits());
+			}
 		}
 	}
 
 	@Test(timeout = 5000)
 	public void testIgnoreError() throws Exception {
 		Exception cause = new RuntimeException();
 		CassandraFailureHandler failureHandler = failure -> Assert.assertEquals(cause, failure);
-		TestCassandraSink casSinkFunc = new TestCassandraSink(failureHandler);
+		try (TestCassandraSink casSinkFunc = new TestCassandraSink(failureHandler)) {
+			casSinkFunc.open(new Configuration());
 
-		casSinkFunc.open(new Configuration());
+			casSinkFunc.enqueueResultSetFuture(ResultSetFutures.fromCompletableFuture(FutureUtils.getFailedFuture(cause)));
+			casSinkFunc.enqueueResultSetFuture(ResultSetFutures.fromCompletableFuture(FutureUtils.getFailedFuture(cause)));
 
-		casSinkFunc.setResultFuture(ResultSetFutures.fromCompletableFuture(FutureUtils.getFailedFuture(cause)));
-
-		casSinkFunc.invoke("hello");
-		casSinkFunc.invoke("world");
+			casSinkFunc.invoke("hello");
+			casSinkFunc.invoke("world");
+		}
 	}
 
 	@Test(timeout = 5000)
 	public void testThrowErrorOnSnapshot() throws Exception {
 		TestCassandraSink casSinkFunc = new TestCassandraSink();
 
-		OneInputStreamOperatorTestHarness<String, Object> testHarness =
-			new OneInputStreamOperatorTestHarness<>(new StreamSink<>(casSinkFunc));
-
-		testHarness.open();
+		try (OneInputStreamOperatorTestHarness<String, Object> testHarness =
+			new OneInputStreamOperatorTestHarness<>(new StreamSink<>(casSinkFunc))) {
+			testHarness.open();
 
-		Exception cause = new RuntimeException();
-		casSinkFunc.setResultFuture(ResultSetFutures.fromCompletableFuture(FutureUtils.getFailedFuture(cause)));
+			Exception cause = new RuntimeException();
+			casSinkFunc.enqueueResultSetFuture(ResultSetFutures.fromCompletableFuture(FutureUtils.getFailedFuture(cause)));
 
-		casSinkFunc.invoke("hello");
+			casSinkFunc.invoke("hello");
 
-		try {
-			testHarness.snapshot(123L, 123L);
+			try {
```

Comment:
I would appreciate if we would move the majority of these changes into a separate commit. (try-catch introduction mostly).